dev/com.ibm.ws.jdbc_fat_v43/test-applications/app43/src/jdbc/fat/v43/web/JDBC43TestServlet.java

```diff
@@ -582,4 +590,152 @@ public void testSerialUnsharedInLTC() throws Exception {
+            PreparedStatement pstmt = con.prepareStatement("INSERT INTO STREETS VALUES(?, ?, ?)");
+            CallableStatement cstmt = con.prepareCall("CALL SYSCS_UTIL.SYSCS_EMPTY_STATEMENT_CACHE()");
+
+            assertEquals("enquotedIdentifier", stmt.enquoteIdentifier("West River Road", false));
+            assertEquals("enquotedIdentifier", pstmt.enquoteIdentifier("IBM's Parking Lot", true));
+            assertEquals("enquotedIdentifier", cstmt.enquoteIdentifier("'41st Street'", true));
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Tests that the new JDBC 4.3 method enquoteNCharLiteral is correctly delegated to the driver on Statement,
+     * PreparedStatement, and CallableStatement
+     */
+    @Test
+    public void testEnquoteNCharLiteral() throws Exception {
+        Connection con = defaultDataSource.getConnection();
+        try {
+            Statement stmt = con.createStatement();
+            PreparedStatement pstmt = con.prepareStatement("INSERT INTO STREETS VALUES(?, ?, ?)");
+            CallableStatement cstmt = con.prepareCall("CALL SYSCS_UTIL.SYSCS_EMPTY_STATEMENT_CACHE()");
+
+            assertEquals("enquotedNCharLiteral", stmt.enquoteNCharLiteral("West River Road"));
+            assertEquals("enquotedNCharLiteral", pstmt.enquoteNCharLiteral("IBM's Parking Lot"));
+            assertEquals("enquotedNCharLiteral", cstmt.enquoteNCharLiteral("'41st Street'"));
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Test that attempting to execute either of the builder methods accessible on the Liberty DataSource
+     * wrapper results in a SQLFeatureNotSupported Exception.
+     */
+    @Test
+    public void testBuilderMethodsNotSuported() throws Exception {
+        try {
+            defaultDataSource.createConnectionBuilder();
+            fail("Call to createConnectionBuilder should result in an exception");
+        } catch (SQLFeatureNotSupportedException ex) {
+            //expected
+        }
+
+        try {
+            defaultDataSource.createShardingKeyBuilder();
+            fail("Call to createShardingKeyBuilder should result in an exception");
+        } catch (SQLFeatureNotSupportedException ex) {
+            //expected
+        }
+    }
+
+    /**
+     * Test that attempting to use the builder methods on an unwrapped connection is blocked.
+     */
+    @Test
+    @ExpectedFFDC({ "java.sql.SQLFeatureNotSupportedException" })
+    public void testBuilderMethodsBlockedUnwrap() throws Exception {
+        XADataSource xaDS = unsharableXADataSource.unwrap(XADataSource.class);
+        try {
+            xaDS.createXAConnectionBuilder();
+            fail("Call to createXAConnectionBuilder on XADataSource should result in an exception");
+        } catch (SQLFeatureNotSupportedException ex) {
+            if (!ex.getMessage().contains("DSRA9130E"))
+                throw ex;
+        }
+
+        try {
```

Comment:
Given that this has nothing to do with unwrap, why is it under this test method?
I expect that when we do the work to support sharding, the mock JDBC driver will implement this method and return an actual sharding key builder, in which case this part of the test will need to be removed anyway.

Reply:
Correct, it shouldn't be there, I removed it when I unblocked the method from above.